.github/workflows/branch-build.yml

```diff
@@ -1,56 +1,61 @@
-name: branch build
+name: PR Build
 
-on:
-  push:
-    branches:
-    - master
-  pull_request:
+on: [pull_request]
 
 jobs:
   build:
     runs-on: ubuntu-latest
     steps:
+
     - name: Setup Node
       uses: actions/setup-node@v1
       with:
         node-version: 14.x
+
     - uses: actions/checkout@v2
       with:
-        fetch-depth: 2
-    - name: Cache .yarn
-      uses: c-hive/gha-yarn-cache@v1
+        fetch-depth: 2 # So packtracker can get the previous commit
+
     - name: Cache node_modules
       uses: actions/cache@v2
       env:
-        cache-name: cache_node_modules
+        cache-name: cache-node_modules
       with:
         path: $GITHUB_WORKSPACE/node_modules
-        key: node_modules-${{ hashFiles('**/yarn.lock')}}
+        key: ${{ runner.os }}-build-${{ env.cache-name }}-${{ hashFiles('**/yarn.lock')}}
         restore-keys: |
-          node_modules-${{ hashFiles('**/yarn.lock')}}
-          node_modules-
-    - name: Cache .cache
+          ${{ runner.os }}-build-${{ env.cache-name }}-${{ hashFiles('**/yarn.lock')}}
+          ${{ runner.os }}-build-${{ env.cache-name }}-
+          ${{ runner.os }}-build-
+          ${{ runner.os }}-
+
+    - name: Cache eslint
       uses: actions/cache@v2
       env:
-        cache-name: cache_home_cache
+        cache-name: cache-eslint
       with:
-        path: $HOME/.cache/
-        key: home-cache-${{ hashFiles('**/yarn.lock')}}
+        path: ~/.cache
+        key: ${{ runner.os }}-build-${{ env.cache-name }}-${{ hashFiles('**/eslint.json') }}
         restore-keys: |
-          home-cache-${{ hashFiles('**/yarn.lock')}}
-          home-cache-
+          ${{ runner.os }}-build-${{ env.cache-name }}-${{ hashFiles('**/eslint.json') }}
+          ${{ runner.os }}-build-${{ env.cache-name }}-
+          ${{ runner.os }}-build-
+          ${{ runner.os }}-
+
     - name: Install
-      run: yarn install --frozen-lockfile --prefer-offline
+      uses: bahmutov/npm-install@v1
+
     - name: Test
       run: yarn test
-    - name: Lint
-      run: yarn run lint-check
-    - if: github.event_name == 'push'
-      name: Set branch env var
-      run: echo "PUSH_BRANCH=${GITHUB_REF##*/}" >> $GITHUB_ENV
-    - name: Build
+
+    - name: Lint check
+      run: |
+        find . -type d -name "node_modules" -prune -o -type f -iregex '.*.ts\|.*.js\|.*.tsx\|.*.jsx' -exec ./build/set-mtime-to-md5.sh {} \;
```

Comment:
This needs to before your cache check for eslint

Reply:
Are you sure? I stole this directly from your previous pr for actions.

Reply:
Nope nevermind